crates/rooch-indexer/src/lib.rs

```diff
@@ -32,31 +32,73 @@ pub type SqlitePoolConnection = diesel::r2d2::PooledConnection<ConnectionManager
 /// Returns all endpoints for which we have implemented on the indexer,
 /// some of them are not validated yet.
 /// NOTE: we only use this for integration testing
-const IMPLEMENTED_METHODS: [&str; 4] = [
-    "multi_get_transactions",
-    "multi_get_events",
-    // indexer apis
-    "query_transactions",
-    "query_events",
-];
-
-pub struct Indexer;
-
-impl Indexer {
-    pub async fn start<S: IndexerStore + Sync + Send + Clone + 'static>(
-        _config: &IndexerConfig,
-        _store: S,
-    ) -> Result<(), IndexerError> {
-        info!("Rooch indexer started...",);
+// const IMPLEMENTED_METHODS: [&str; 4] = [
+//     "multi_get_transactions",
+//     "multi_get_events",
+//     // indexer apis
+//     "query_transactions",
+//     "query_events",
+// ];
+
+#[derive(Clone)]
+pub struct IndexerStore {
+    pub sqlite_store: SqliteIndexerStore,
+}
 
-        Ok(())
+impl IndexerStore {
+    pub fn new(db_url: &str) -> Result<Self> {
+        let sqlite_cp = new_sqlite_connection_pool(db_url)?;
+        let store = Self {
+            sqlite_store: SqliteIndexerStore::new(sqlite_cp),
+        };
+        Ok(store)
+    }
+
+    pub fn mock_indexer_store() -> Result<Self> {
+        let tmpdir = moveos_config::temp_dir();
```

Comment:
Should keep the `temp_dir()` returned `DataDirPath`. Otherwise, the temp dir will be removed when the DataDirPath instance is dropped.

Reply:
Fixed the problem of using temp path to release instance when server starts.
As for whether calling `mock_indexer_store` in the integration test will release it, I will write a test case to cover and verify it later.